source/extensions/filters/http/transformation/transformation_filter.cc

```diff
@@ -213,6 +220,15 @@ void TransformationFilter::transformSomething(
     Buffer::Instance &body, void (TransformationFilter::*responeWithError)(),
     void (TransformationFilter::*addData)(Buffer::Instance &)) {
 
+  // if there is no route level config, and there are header matchers, check them
+  if (!has_route_level_config_ && !filter_config_->matchHeaders(*request_headers_)) {
```

Comment:
we should probably do this logic in checkRequestActive so that the filter will be a NOP if no action is needed